auth/handler.go

```diff
@@ -182,6 +182,18 @@ func (h *Handler) EndAllUserSessionsTx(ctx context.Context, tx *sql.Tx) error {
 	return err
 }
 
+func (h *Handler) FindUserIDForAuthSubject(ctx context.Context, providerID, subjectID string) (string, error) {
+	var userID string
+	err := h.userLookup.QueryRowContext(ctx, providerID, subjectID).Scan(&userID)
+	if errors.Is(err, sql.ErrNoRows) {
+		err = nil
+	}
+	if err != nil {
+		return "", err
+	}
+	return userID, nil
+}
```

Comment:
This has no input validation and should be handled in the engine in a similar way to how we use a phone number to map to a user